exporter/common_test.go

```diff
@@ -8,65 +8,127 @@ import (
+		allDBs, err := databases(ctx, client, nil, systemDBs)
+		assert.NoError(t, err)
+		assert.Equal(t, want, allDBs)
+	})
 
-	// List all collections in testdb01 (inDBs[0]) but only col01 and col02 from testdb02.
-	filterInNameSpaces := []string{inDBs[0], inDBs[1] + ".col01", inDBs[1] + ".col02"}
-	namespaces, err := listAllCollections(ctx, client, filterInNameSpaces, systemDBs)
-	assert.NoError(t, err)
-	assert.Equal(t, wantNS, namespaces)
+	t.Run("One collection in list", func(t *testing.T) {
+		want := []string{"testdb01"}
+		allDBs, err := databases(ctx, client, []string{"testdb01.col1"}, systemDBs)
+		assert.NoError(t, err)
+		assert.Equal(t, want, allDBs)
+	})
+
+	t.Run("Multiple namespaces in list", func(t *testing.T) {
+		want := []string{"testdb01", "testdb02"}
+		allDBs, err := databases(ctx, client, []string{"testdb01", "testdb02.col2", "testdb02.col1"}, systemDBs)
+		assert.NoError(t, err)
+		assert.Equal(t, want, allDBs)
+	})
+}
+
+//nolint:paralleltest
+func TestListCollections(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	client := tu.DefaultTestClient(ctx, t)
+
+	setupDB(ctx, t, client)
+	defer cleanupDB(ctx, client)
+
+	t.Run("Basic test", func(t *testing.T) {
+		want := []string{"admin", "config", "local", "testdb01", "testdb02"}
+		allDBs, err := databases(ctx, client, nil, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, want, allDBs)
+	})
+
+	t.Run("Filter in databases", func(t *testing.T) {
+		want := []string{"col01", "col02", "colxx"}
+		inNameSpaces := []string{testDBs[0] + ".col0", testDBs[0] + ".colx"}
+		colls, err := listCollections(ctx, client, testDBs[0], inNameSpaces)
+		sort.Strings(colls)
+
+		assert.NoError(t, err)
+		assert.Equal(t, want, colls)
+	})
+
+	t.Run("With namespaces list", func(t *testing.T) {
+		// Advanced filtering test
+		wantNS := map[string][]string{
+			"testdb01": {"col01", "col02", "colxx", "colyy"},
+			"testdb02": {"col01", "col02"},
+		}
+		// List all collections in testdb01 (inDBs[0]) but only col01 and col02 from testdb02.
+		filterInNameSpaces := []string{testDBs[0], testDBs[1] + ".col01", testDBs[1] + ".col02"}
+		namespaces, err := listAllCollections(ctx, client, filterInNameSpaces, systemDBs)
+		assert.NoError(t, err)
+		assert.Equal(t, wantNS, namespaces)
+	})
+
+	t.Run("Empty namespaces list", func(t *testing.T) {
+		wantNS := map[string][]string{
+			"testdb01": {"col01", "col02", "colxx", "colyy"},
+			"testdb02": {"col01", "col02", "colxx", "colyy"},
+		}
+		namespaces, err := listAllCollections(ctx, client, nil, systemDBs)
+		assert.NoError(t, err)
+		assert.Equal(t, wantNS, namespaces)
+	})
 
-	count, err := nonSystemCollectionsCount(ctx, client, nil, nil)
-	assert.NoError(t, err)
-	assert.True(t, count == 8)
+	t.Run("Count basic", func(t *testing.T) {
+		count, err := nonSystemCollectionsCount(ctx, client, nil, nil)
+		assert.NoError(t, err)
+		assert.True(t, count == 8)
```

Comment:
Inconsistent assertion here and [line 132](https://github.com/percona/mongodb_exporter/pull/402/files#diff-2c516d0b4714086d6ac8c4d7ae0e5929c80150a06e0fb4d2ed543906dbd2c866R132). Prefer to use `assert.Equal`